src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,9 +32,24 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
-  private readTasks(): Task[] {
+  private readTasks(query?:string): Task[] {
+
     const tasks = localStorage.getItem(LocalTaskService.STORAGE_KEY);
-    return tasks ? JSON.parse(tasks) : [];
+    const tasksList = JSON.parse(tasks);
+    const searchResult = [];
+
+    if (tasks && !query) {
+      return tasksList;
```

Comment:
by having a dedicated method for searching, this if branch wouldn't be necessary as you determine right away by the method call in the` app.component.ts` that you would have a valid query to search for